src/TraceEvent/Computers/ContentionTimeComputer.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System.Collections.Generic;
+using Microsoft.Diagnostics.Tracing.Etlx;
+using Microsoft.Diagnostics.Tracing.Parsers.Clr;
+using Microsoft.Diagnostics.Tracing.Stacks;
+
+namespace Microsoft.Diagnostics.Tracing.Computers
+{
+    public class ContentionTimeComputer
+    {
+        private readonly TraceLog _eventLog;
+        private readonly MutableTraceEventStackSource _stackSource;
+        private readonly Dictionary<int, StackSourceSample> _samplesPerThread;
+
+        private const double NanosInMillisecond = 1000 * 1000;
+
+        public ContentionTimeComputer(TraceLog eventLog, MutableTraceEventStackSource stackSource)
+        {
+            _eventLog = eventLog;
+            _stackSource = stackSource;
+            _samplesPerThread = new Dictionary<int, StackSourceSample>();
+        }
+
+        public void GenerateContentionTimeStacks()
+        {
+            var eventSource = _eventLog.Events.GetSource();
+            
+            eventSource.Clr.ContentionStart += OnContentionStart;
+            eventSource.Clr.ContentionStop += OnContentionStop;
+
+            eventSource.Process();
+            
+            _stackSource.DoneAddingSamples();
+        }
+
+        private void OnContentionStart(ContentionStartTraceData startData)
+        {
+            if (!_samplesPerThread.TryGetValue(startData.ThreadID, out var sample))
+            {
+                _samplesPerThread[startData.ThreadID] = sample = new StackSourceSample(_stackSource);
```

Comment:
Instead of using `startData.ThreadID` here, please use `startData.Thread().ThreadIndex`.  This will handle the case where thread IDs get reused without you having to keep track of process/thread creation/termination.